### PR TITLE
dev/core#2719 [REF] Deprecate legacy references to contribution_invoice_settings

### PR DESCRIPTION
**Pull request by eileenmcnaughton**
Overview
----------------------------------------
dev/core#2719 [REF] Deprecate legacy references to contribution_invoice_settings

Before
----------------------------------------
`Civi::settings()->get('contribution_invoice_settings')` is a silent wrapper for settings that used to be subkeys of it

After
----------------------------------------
The above is a noisy wrapper 

Technical Details
----------------------------------------
Time to get rid of this old cruft - I expect a couple of test fails

Comments
----------------------------------------
per #20862 discussion with @mlutfy

### CODE REVIEW

File: Civi/Core/SettingsBag.php [hunk 1]
```diff
@@ -182,6 +182,9 @@ public function all() {
    */
   public function get($key) {
     $all = $this->all();
+    if ($key === 'contribution_invoice_settings') {
+      \CRM_Core_Error::deprecatedWarning('Invoicing settings should be directly accessed - eg Civi::setting()->get("invoicing")');
```
Comment (demeritcowboy) [inline Civi/Core/SettingsBag.php:186]: This is a typo I've made myself 18 times - should be Civi::settings not Civi::setting.

**Comment by seamuslee001** [comment]
Just a <sarcasm>few</sarcasm> test failures

**Comment by eileenmcnaughton** [comment]
Yeah - this one will affect a lot of code paths -https://github.com/civicrm/civicrm-core/pull/20971

**Comment by eileenmcnaughton** [comment]
test this please

**Comment by eileenmcnaughton** [comment]
the tests are finishing now.... https://github.com/civicrm/civicrm-core/pull/20991 should clear out a bunch more

**Comment by eileenmcnaughton** [comment]
test this please

**Comment by eileenmcnaughton** [comment]
I'm gonna close this - it's blocked by https://github.com/civicrm/civicrm-core/pull/21120 at the moment but may come back to life one day

**Comment by eileenmcnaughton** [comment]
@colemanw - this is passing now - time to get those warnings happeing...

**Comment by colemanw** [comment]
Nice resurrection!

**Pull request merged**